csharp/src/Ice/InputStream.cs

```diff
@@ -1127,7 +1127,7 @@ internal Endpoint ReadEndpoint(Protocol protocol)
             }
             else
             {
-                endpoint = Endpoint.FromEndpointData(new EndpointData(this), Communicator, protocol);
+                endpoint = (new EndpointData(this)).ToEndpoint(Communicator, protocol);
```

Comment:
Don't need the extra parenthesis here
```suggestion
                endpoint = new EndpointData(this).ToEndpoint(Communicator, protocol);
```

Reply:
Fixed.